From ebcd693830ea91a4fe925ab72ee4f94ab972b6c0 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@st.com>
Date: Fri, 21 Oct 2022 11:39:31 +0200
Subject: [PATCH 048/841] feat(stm32mp15-fdts): add stm32mp157f-dk2 board

This commit adds stm32mp157f-dk2 boards support. This board embeds a
STM32MP157F SOC. This SOC contains the same level of feature than a
STM32MP157C SOC but A7 clock frequency can reach 800MHz.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git acd49fe349187effda065c9206c2449d15aa2200 feat(stm32mp15-fdts): add stm32mp157f-dk2 board]
Signed-off-by: Alexandre Torgue <alexandre.torgue@foss.st.com>
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Change-Id: Ic12dff303cc8feff3df4fbeefd1347525e2bcd0a
---
 fdts/stm32mp157f-dk2.dts | 33 +++++++++++++++++++++++++++++++++
 1 file changed, 33 insertions(+)
 create mode 100644 fdts/stm32mp157f-dk2.dts

diff --git a/fdts/stm32mp157f-dk2.dts b/fdts/stm32mp157f-dk2.dts
new file mode 100644
index 000000000..bf88c6df1
--- /dev/null
+++ b/fdts/stm32mp157f-dk2.dts
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
+ * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
+ */
+
+/dts-v1/;
+
+#include "stm32mp157.dtsi"
+#include "stm32mp15xf.dtsi"
+#include "stm32mp15-pinctrl.dtsi"
+#include "stm32mp15xxac-pinctrl.dtsi"
+#include "stm32mp15xx-dkx.dtsi"
+
+/ {
+	model = "STMicroelectronics STM32MP157F-DK2 Discovery Board";
+	compatible = "st,stm32mp157f-dk2", "st,stm32mp157";
+
+	aliases {
+		serial0 = &uart4;
+		serial1 = &usart3;
+		serial2 = &uart7;
+		serial3 = &usart2;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+};
+
+&cryp1 {
+	status = "okay";
+};
-- 
2.47.2

